api/v1alpha1/virtualmachineimage_types.go

```diff
@@ -78,6 +78,10 @@ type VirtualMachineImageSpec struct {
 	// OVFEnv describes the user configurable customization parameters of the VirtualMachineImage.
 	// +optional
 	OVFEnv map[string]OvfProperty `json:"ovfEnv,omitempty"`
+
+	// HardwareVersion describes the virtual hardware version of the image
+	// +optional
+	HardwareVersion int32 `json:"hwVersion,omitempty"`
```

Comment:
To make this optional, shouldn't this be a `*int32` (otherwise unset value would equal 0)

Reply:
I had that question too. I found out that when the value is unset 0 (ie) empty for an int32, it gets omitted from describe/get for the virtualmachineimage object. And since we'll be the only consumers right now, All checks in vmoperator look for unset val 0 before making any decisions based on it.

Reply:
Ah okay, we can keep it as is, thanks for checking